substrate/frame/transaction-payment/src/lib.rs

```diff
@@ -830,6 +774,32 @@ where
 	}
 }
 
+/// Require the transactor pay for themselves and maybe include a tip to gain additional priority
+/// in the queue.
+///
+/// # Transaction Validity
+///
+/// This extension sets the `priority` field of `TransactionValidity` depending on the amount
+/// of tip being paid per weight unit.
+///
+/// Operational transactions will receive an additional priority bump, so that they are normally
+/// considered before regular transactions.
+#[derive(Encode, Decode, Clone, Eq, PartialEq, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub struct ChargeTransactionPayment<T: Config>(#[codec(compact)] BalanceOf<T>);
```

Comment:
no changes to the API of this type. `get_priority` method decoupled to the separate `Priority` type to be reused. the `withdraw_fee` and `can_withdraw_fee` just dropped.